.circleci/config.yml

```diff
@@ -42,6 +42,21 @@ jobs:
     <<: *defaults
     steps:
       - *restore_repo
+      - run:
+          name: Check if yarn prettier was run
+          command: |
+            # Files changed on this branch
+            CHANGED_FILES=$(git diff --name-only master...)
+            # Files that should have been formatted while working on this branch
+            FORMATTED_FILES=$(yarn --silent prettier:files | grep "$CHANGED_FILES")
```

Comment:
Do we run prettier twice here? Would it be faster to run it once and to execute `git diff --exit-code`?
Also, I think that we scope prettier to the TypeScript files :).

Reply:
`find` always did pipe `*.d.ts` and `*.tsx` to the `prettier` script. `eslint` only processes `jsx?` files though as far as I know. 

I was never really a fan of `eslint-plugin-prettier`. Mostly this comes down to personal preference but also some actual minor issues: `eslint` has no parser for `ts` files which makes `eslint --fix` with `eslint-plugin-prettier` useless for ts files. `eslint-plugin-prettier` allows `prettier` configuration with `.eslintrc` which is [ignored by ide plugins](https://github.com/prettier/eslint-plugin-prettier#options).

`prettier:files` is just a helper method at the moment which lists files that should be formatted. It is not actually running `prettier --write`.

`prettier --write` should never be used in a CI environment. That's the hole purpose of [`--list-different`](https://prettier.io/docs/en/cli.html#list-different). Combine `--write` with `git diff` while reducing I/O operations.

It was intended that the new check enforces a consistent code style across ts and js file. Just like the existing `prettier` scripts already formats `tsx?` files.

Reply:
Now I understood what you mean. Since we run prettier via `eslint` we would run it here again on all `jsx?` files. I removed `eslint-plugin-prettier` to removed this redundancy.

Reply:
> eslint --fix

I never used that. I have never been satisfied with the output.

> I was never really a fan of eslint-plugin-prettier

Most people configure their editor to run eslint. The advantage of this plugin is that you know up front that you need to run prettier, it's just like another eslint rule. At least, it's my workflow. I don't automatically run prettier on save. This data is interesting: https://npm-stat.com/charts.html?package=eslint-plugin-prettier&package=prettier.

Reply:
Well prettier supports more than `jsx?` while eslint only runs on `jsx?`. Typescript support might explain some difference.

I didn't think about editor integration though and while I also don't use formatOnSave I do frequently press the format hotkey.